DSCResources/MSFT_xVHD/MSFT_xVHD.psm1

```
@@ -75,6 +80,17 @@ function Set-TargetResource
         Throw "Please ensure that Hyper-V role is installed with its PowerShell module"
     }
 
+    # Check the module combinations make sense
+    if($ParentPath -and $Type -ne "Differencing")
+    {
+        Throw "Parent path is only supported for Differencing disks"
+    }
+
+    if($null -eq $ParentPath -and $Type -eq "Differencing")
+    {
+        Throw "Differencing requires a parent path"
```

And It 'Should throw when ParentPath is not defined and type is Differencing'